ts/src/poloniex.ts

```diff
@@ -677,36 +683,94 @@ export default class poloniex extends Exchange {
             const id = this.safeValue (ids, 0);
             const currency = this.safeValue (item, id);
             const code = this.safeCurrencyCode (id);
+            const name = this.safeString (currency, 'name');
+            const networkId = this.safeString (currency, 'blockchain');
+            const networkCode = this.networkIdToCode (networkId, code);
             const delisted = this.safeValue (currency, 'delisted');
-            const walletState = this.safeString (currency, 'walletState');
-            const enabled = walletState === 'ENABLED';
-            const listed = !delisted;
-            const active = listed && enabled;
+            const walletEnabled = this.safeString (currency, 'walletState') === 'ENABLED';
+            const depositEnabled = this.safeString (currency, 'walletDepositState') === 'ENABLED';
+            const withdrawEnabled = this.safeString (currency, 'walletWithdrawalState') === 'ENABLED';
+            const active = !delisted && walletEnabled && depositEnabled && withdrawEnabled;
             const numericId = this.safeInteger (currency, 'id');
-            const fee = this.safeNumber (currency, 'withdrawalFee');
-            result[code] = {
-                'id': id,
-                'numericId': numericId,
-                'code': code,
+            const feeString = this.safeString (currency, 'withdrawalFee');
+            const parentChain = this.safeValue (currency, 'parentChain');
+            const noParentChain = parentChain === undefined;
```

Comment:
@ProfHollander I was testing it and couldn't find any `parentChain`, so I don't think we need this logic

Reply:
@carlosmiei sorry, my bad. Forgot to change request to poloniex API.
When we call method `publicGetCurrencies` without params we get response without `parentChain` ( https://api.poloniex.com/currencies ). And for one currency we get only one chain in response (only parentChain). 
For example for USDT in response there will only one parent OMNI chain:
```
"USDT" : {
    "id" : 214,
    "name" : "Tether USD",
    "description" : "Sweep to Main Account",
    "type" : "address",
    "withdrawalFee" : "0.00000000",
    "minConf" : 2,
    "depositAddress" : null,
    "blockchain" : "OMNI",
    "delisted" : false,
    "tradingState" : "NORMAL",
    "walletState" : "DISABLED",
    "walletDepositState" : "DISABLED",
    "walletWithdrawalState" : "DISABLED",
    "supportCollateral" : true,
    "supportBorrow" : true
  }
```
  
But if we call another the same method but will extend it with additional params ( https://api.poloniex.com/currencies?includeMultiChainCurrencies=true ) in response  we will get all chains of this currency and the response will be a little bit different. For example for USDT the response will be for main chain OMNI:
```
"USDT" : {
    "id" : 214,
    "name" : "Tether USD",
    "description" : "Sweep to Main Account",
    "type" : "address",
    "withdrawalFee" : "0.00000000",
    "minConf" : 2,
    "depositAddress" : null,
    "blockchain" : "OMNI",
    "delisted" : false,
    "tradingState" : "NORMAL",
    "walletState" : "DISABLED",
    "walletDepositState" : "DISABLED",
    "walletWithdrawalState" : "DISABLED",
    "supportCollateral" : true,
    "supportBorrow" : true,
    "parentChain" : null,
    "isMultiChain" : true,
    "isChildChain" : false,
    "childChains" : [ "USDTBSC", "USDTETH", "USDTTRON" ]
  }
```
  
  And we can see several `childChains`, for example:
  
```
 "USDTETH" : {
    "id" : 318,
    "name" : "USDT on ETH",
    "description" : "Sweep to Main Account",
    "type" : "address",
    "withdrawalFee" : "4.10685474",
    "minConf" : 64,
    "depositAddress" : null,
    "blockchain" : "ETH",
    "delisted" : false,
    "tradingState" : "NORMAL",
    "walletState" : "ENABLED",
    "walletDepositState" : "ENABLED",
    "walletWithdrawalState" : "ENABLED",
    "supportCollateral" : false,
    "supportBorrow" : false,
    "parentChain" : "USDT",
    "isMultiChain" : true,
    "isChildChain" : true,
    "childChains" : [ ]
  }
```
  
  I fixed the code already and made new commit.

Reply:
@ProfHollander, thanks will check it again